Dispatcher/UI/Auth/AuthViewController.swift

```diff
@@ -7,57 +7,196 @@
+            emailTextField?.layer.borderWidth = 0
+            emailErrorLabel.text = ""
+            
+            passwordTextField?.styleTextFieldPlaceHolder(placeholderText: TextCostants.passwordTextFieldPlaceholder, fontColor: UIColor(named: ColorsPalleteNames.labelsTextColor))
+            passwordTextField?.textColor = UIColor(named: ColorsPalleteNames.labelsTextColor)
+            passwordTextField?.text = nil
+            passwordTextField?.layer.borderWidth = 0
+            passwordErrorLabel.text = ""
+            
+            reEnterPasswordTextField?.styleTextFieldPlaceHolder(placeholderText: TextCostants.reEnterPasswordTextFieldPlaceholder, fontColor: UIColor(named: ColorsPalleteNames.labelsTextColor))
+            reEnterPasswordTextField?.textColor = UIColor(named: ColorsPalleteNames.labelsTextColor)
+            reEnterPasswordTextField?.text = nil
+            reEnterPasswordTextField?.layer.borderWidth = 0
+            reEnterPasswordTextField?.text = ""
+            reEnterPasswordErrorLabel.text = ""
+        }
+        
+        private func styleErrorLabel(label: UILabel, error: String) {
+            label.text = error
+            label.textColor = UIColor(named: ColorsPalleteNames.errorColor)
+            label.isHidden = false
         }
-    }
     
-    func signupOrLoginPressed(authMode: AuthMode, email: String, password: String) {
-        authViewModel.authenticateUser(authMode: authMode, email: email, password: password, completionHandler: { errorMsg in
-            if(errorMsg != nil) {
-                self.present(createErrorAlert(errorMsg!), animated: true, completion: nil)
+        @IBAction func mainButtonPressed() {
+            if (validatedEmail == "" || validatedPassword == "") {
+                if (validatedEmail == "") {
+                    emailTextField?.styleErrorTextField()
+                    styleErrorLabel(label: emailErrorLabel, error: UserInputErrors.invalidEmailError.errorDescription ?? "")
+                } else if (validatedPassword == "") {
+                    passwordTextField?.styleErrorTextField()
+                    styleErrorLabel(label: passwordErrorLabel, error: UserInputErrors.invalidPasswordError.errorDescription ?? "")
+                }
             } else {
-                self.performSegue(withIdentifier: SegueIdentifiers.fromAuthToTabBar, sender: self)
+                authViewModel.authenticateUser(authMode: authMode, email: validatedEmail, password: validatedPassword) { errorMsg in
+                    if let err = errorMsg {
+                        self.present(createErrorAlert(err), animated: true, completion: nil)
+                    } else {
+                        self.performSegue(withIdentifier: SegueIdentifiers.fromAuthToTabBar, sender: self)
+                    }
+                }
             }
-        })
+        }
+        
+        @IBAction func secondaryButtonPressed() {
+            if (authMode == AuthMode.Login) {
```

Comment:
Can shorten this to a one-liner like:
authMode = (authMode == AuthMode.Login) ? AuthMode.Signup : AuthMode.Login